lib/streamlit/elements/widgets/number_input.py

```diff
@@ -194,7 +194,9 @@ def number_input(
         format : str or None
             A printf-style format string controlling how the interface should
             display numbers. Output must be purely numeric. This does not impact
-            the return value. Valid formatters: %d %e %f %g %i %u
+            the return value. Formatting is handled by [sprintf.js](https://github.com/alexei/sprintf.js).
+            This can be used to adjust decimal precision in the displayed result. For example,
+            ``'%0.1f'`` to only show 1 digit after the decimal.
```

Comment:
Tagging @sfc-gh-dmatthews just to inform in case she has any comments, but it lgtm.